PolyEngine/Core/Src/pe/core/utils/Result.hpp

```diff
@@ -10,20 +10,39 @@ namespace pe::core::utils
 	{
 	public:
 		Result() = delete;
-		Result(Result&) = delete;
+		Result(Result&) = default;
 		Result(Result&&) = default;
-		Result& operator=(Result&) = delete;
+		Result& operator=(Result&) = default;
 		Result& operator=(Result&&) = default;
 		Result(T okVal) : m_value(std::move(okVal)) {}
 		Result(E errVal) : m_value(std::move(errVal)) {}
 		
-		bool IsOk() const { return m_value.index() == 0; }
-		bool IsErr() const { return !IsOk(); }
-		T& GetValue() { return std::get<T>(m_value); }
-		E& GetError() { return std::get<E>(m_value); }
-		const T& GetValue() const { return std::get<T>(m_value); }
-		const E& GetError() const { return std::get<E>(m_value); }
-	
+		bool isOk() const { return m_value.index() == 0; }
+		bool isErr() const { return !isOk(); }
+		
+		T& getValue() { return std::get<T>(m_value); }
+		E& getError() { return std::get<E>(m_value); }
```

Comment:
maybe T& getDefault(T&) (returning default if error) and
T& getHandle(T&(E&)) would be usefull to add?

Reply:
I suggest different naming scheme:
`value()`
`error()`
`valueOr(T)`

What `getHandle` is supposed to do?

Reply:
like `getDefault`/`valueOr` but uses function taking error and returning default value instead.

Reply:
done